net-libs/miniupnpc/miniupnpc-2.0.20180222.recipe

```diff
@@ -0,0 +1,73 @@
+SUMMARY="UPnP IGD client lightweight library"
+DESCRIPTION="The usage of the miniUPnP client library is useful whenever an \
+application needs to listen for incoming connections.
+Examples : P2P applications, FTP clients for active mode, IRC (for DCC)
+or IM applications, network games, any server."
+HOMEPAGE="http://miniupnp.free.fr"
+COPYRIGHT="2005-2018 Thomas Bernard"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="http://miniupnp.free.fr/files/miniupnpc-$portVersion.tar.gz"
+CHECKSUM_SHA256='587944686469d09f739744b3aed70c7ce753a79c40d6f1227f68a3e962665b75'
+PATCHES="miniupnpc-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	miniupnpc$secondaryArchSuffix = $portVersion
+	lib:libminiupnpc$secondaryArchSuffix = 2.0.0 compat >= 2
```

Comment:
The package produces a `libminiupnpc.so.16 -> libminiupnpc.so.2.0` symlink, so maybe we should make this change:
```
-	lib:libminiupnpc$secondaryArchSuffix = 2.0.0 compat >= 2
+	lib:libminiupnpc$secondaryArchSuffix = 2.0 compat >= 16
```
but I don't know if this would be OK for haikuporter.

Same thing in `PROVIDES_devel`: 
```
-	devel:libminiupnpc$secondaryArchSuffix = 2.0.0 compat >= 2
+	devel:libminiupnpc$secondaryArchSuffix = 2.0 compat >= 16
```

So maybe we should use `2.0 compat >= 2` (instead of `2.0.0 compat >= 2`)

Reply:
It should be defined based on the soname. `readelf -d` got me this:

    0x000000000000000e (SONAME)             Library soname: [libminiupnpc.so.16]

So yes, the changes you suggested should be done.

Actually, as a matter of fact, if you use the `Makefile` [provided](https://github.com/miniupnp/miniupnp/blob/4a4269a673d6a779143387dca9e70c79e057c74e/miniupnpc/Makefile) in the provided distribution (idk why we used `cmake`), `libminiupnpc.so.16` is the only library that's created after the build. So

    lib:libminiupnpc$secondaryArchSuffix = 16

should be the only thing necessary.